NotificationsMessage.py

```diff
@@ -0,0 +1,50 @@
+from datetime import datetime, timedelta
+import uuid
+
+class NotificationsPipelineHistory(object): # Pipeline History Object
+    def __init__(self):
+        self.enter_timestamp = None
+        self.exit_timestamp = None
+    def onEnter(self):
+        """
+        Update timestamp for enter.
+        """
+        self.enter_timestamp = datetime.utcnow()
```

Comment:
Despite the name, datetime.utcnow() creates a timezone-naive datetime object.  If we're okay with a django dependency, we should use `django.utils.timezone.now()`, otherwise we should use `pytz.utc.localize(datetime.utcnow())`.  Either of those will create a timezone-aware datetime object.